test/upgrade/test_main.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"os"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/eventing-kafka/test"
+	eventingTest "knative.dev/eventing/test"
+	testlib "knative.dev/eventing/test/lib"
+	"knative.dev/eventing/test/lib/resources"
+	"knative.dev/pkg/system"
+)
+
+var (
+	channelTestRunner  testlib.ComponentsTestRunner
+	defaultChannelType = metav1.TypeMeta{
+		APIVersion: resources.MessagingAPIVersion,
```

Comment:
upstream uses `messaging.knative.dev/v1beta1` there - so currently that is cool here.

But... if that changes to `v1`  we are still cool ?   (we dont have a KafkaChannel v1 yet)

Reply:
Hmm. This should be set to a value that is correct for latest released Eventing Kafka, and then HEAD would introduce migration of this to `v1`?

Reply:
the line:

```
		APIVersion: resources.MessagingAPIVersion,

```

points to the vendored code, see: https://github.com/knative/eventing/blob/main/test/lib/resources/constants.go#L28

I wonder... when the eventing repo changes that line to `v1`... what happens here?

Reply:
It will update here, when you update dependencies. And then the upgrade test would probably fail, as it will try to deploy KafkaChannel v1 that doesn't exist. That's my guess.

For me that's okay. When it happens we'll just update this code here to point to correct version. WDYT?

Reply:
but.. if we continue... using that... aren't we than forced to do v1 as soon as upstream (knative eventing) does ?